.github/workflows/build-accumulo.yml

```diff
@@ -0,0 +1,107 @@
+    permissions:
+      contents: read
+      packages: write
+      #
+    steps:
+    - name: Checkout DataWave
+      uses: actions/checkout@v4
+      with:
+        path: datawave
+
+    - name: Checkout Accumulo
+      uses: actions/checkout@v4
+      id: accumuloCheckout
+      with:
+        repository: ${{ github.event.inputs.accumuloRepo }}
+        path: accumulo
+        ref: ${{ github.event.inputs.accumuloBranch }}
+
+    - name: Set up JDK ${{env.ACCUMULO_JAVA_VERSION}}
+      uses: actions/setup-java@v4
+      with:
+        distribution: ${{env.JAVA_DISTRIBUTION}}
+        java-version: ${{env.ACCUMULO_JAVA_VERSION}}
+        cache: 'maven'
+    - uses: actions/cache@v4
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-build-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-build-
+          ${{ runner.os }}-maven-format-
+          ${{ runner.os }}-maven-
+    - run: echo "ACCUMULO_JAVA=$JAVA_HOME" >> $GITHUB_ENV
+    - name: Set up JDK ${{env.DATAWAVE_JAVA_VERSION}}
+      uses: actions/setup-java@v4
+      with:
+        distribution: ${{env.JAVA_DISTRIBUTION}}
+        java-version: ${{env.DATAWAVE_JAVA_VERSION}}
+        cache: 'maven'
+    - run: echo "DATAWAVE_JAVA=$JAVA_HOME" >> $GITHUB_ENV
+
+    - name: Get Accumulo Version
+      id: get-accumulo-version
+      run: |
+        export JAVA_HOME=$ACCUMULO_JAVA
+        cd $GITHUB_WORKSPACE/accumulo
+        mvn build-helper:parse-version versions:set -DgenerateBackupPoms=false -DnewVersion=\${parsedVersion.majorVersion}.\${parsedVersion.minorVersion}.\${parsedVersion.incrementalVersion}-dwv-$(git rev-parse --short HEAD)
+        export newVersion=$(mvn -q help:evaluate -DforceStdout -Dexpression=project.version)
+        echo accumuloVersion=$newVersion >> $GITHUB_OUTPUT
+    - name: Deploy Accumulo
+      if: ${{ ( github.event.inputs.deployAccumulo || 'false' )  == 'true'}}
+      run: | 
+        export JAVA_HOME=$ACCUMULO_JAVA
+        cd $GITHUB_WORKSPACE/accumulo   
+        mvn -DaltDeploymentRepository=github-datawave::https://maven.pkg.github.com/NationalSecurityAgency/datawave -V -B -e -ntp "-Dstyle.color=always" -DskipTests -T1C clean source:jar deploy -s $GITHUB_WORKSPACE/datawave/.github/workflows/settings.xml
+    - name: Log in to the Container registry
+      uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ env.USER_NAME }}
+        password: ${{ env.ACCESS_TOKEN }}
+
+    - name: Update DataWave Dependency Version
+      run: |
+
+        export JAVA_HOME=$DATAWAVE_JAVA
+        cd $GITHUB_WORKSPACE/datawave
+        mvn -s $GITHUB_WORKSPACE/datawave/.github/workflows/settings.xml versions:set-property -Dproperty=version.accumulo -DnewVersion=${{ steps.get-accumulo-version.outputs.accumuloVersion }} -DgenerateBackupPoms=false
+    - name: Build Web and Ingest Docker Images (Maven)
```

Comment:
Does running the datawave build make any changes outside of the build containter?  If it does not then is the purpose of building datewave here just informational, to let someone know if the current version of datawave build against the accumulo snapshot? Related to that, is the intended use of setting this to deployAccumulo to false to see if the current version of datawave builds against the specified accumulo version?

Reply:
The datawave build builds the datawave ingest and web docker images. The ingest image is based on the datawave-accumulo image. Running this build after running a build of the accumulo image in datawave-stack-docker-images ensures that we update the version of accumulo being used for datawave-helm-charts

deployAccumulo exists because the mvn deploy will fail if this version has already been deployed. If the accumulo deploy passes but the datawave build fails, we can update datawave code and rerun with the same accumulo sha without having to delete the packages, as long as deployAccumulo is set to false